app/code/core/Mage/Adminhtml/controllers/Catalog/ProductController.php

```diff
@@ -277,6 +277,13 @@ public function editAction()
             $block->setStoreId($product->getStoreId());
         }
 
+        if ($this->getRequest()->getParam('popin')) {
+            $root = $this->getLayout()->getBlock('root');
+            if ($root) {
+                $root->addBodyClass('popup-nohead-nofoot');
```

Comment:
I don't understand PHPStan here...

Reply:
```$this->getLayout()->getBlock('root');``` returns Mage_Core_Block_Abstract, but ```addBodyClass()``` belongs to a child class ... this should work:

```
            /** @var Mage_Adminhtml_Block_Page $root */
            $root = $this->getLayout()->getBlock('root');
            if ($root) {
```